ncclient/transport/unixSocket.py

```diff
@@ -0,0 +1,89 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import socket
+import sys
+import threading
+
+from socket import AF_UNIX, SOCK_STREAM
+
+if sys.version < '3':
+    from six import StringIO
+else:
+    from io import BytesIO as StringIO
+
+from ncclient.capabilities import Capabilities
+from ncclient.logging_ import SessionLoggerAdapter
+from ncclient.transport.errors import UnixSocketError
+from ncclient.transport.session import Session
+from ncclient.transport.parser import DefaultXMLParser
+
+logger = logging.getLogger("ncclient.transport.unix")
+
+DEFAULT_SOCKET_PATH = "/tmp/socket.sock"
```

Comment:
I think it would be better if there were no default socket path. This could become an inadvertent attack vector.